pkg/logging/logger.go

```diff
@@ -20,38 +20,19 @@ import (
 	"context"
 
 	"go.uber.org/zap"
-)
 
-type loggerKey struct{}
-
-// This logger is used when there is no logger attached to the context.
-// Rather than returning nil and causing a panic, we will use the fallback
-// logger. Fallback logger is tagged with logger=fallback to make sure
-// that code that doesn't set the logger correctly can be caught at runtime.
-var fallbackLogger *zap.SugaredLogger
-
-func init() {
-	if logger, err := zap.NewProduction(); err != nil {
-		// We failed to create a fallback logger. Our fallback
-		// unfortunately falls back to noop.
-		fallbackLogger = zap.NewNop().Sugar()
-	} else {
-		fallbackLogger = logger.Named("fallback").Sugar()
-	}
-}
+	"github.com/knative/pkg/logging"
+)
 
 // WithLogger returns a copy of parent context in which the
 // value associated with logger key is the supplied logger.
 func WithLogger(ctx context.Context, logger *zap.SugaredLogger) context.Context {
-	return context.WithValue(ctx, loggerKey{}, logger)
+	return logging.WithLogger(ctx, logger)
```

Comment:
Is the eventual goal to remove these trampolines and replace calls to `knative/serving/pkg/logging.WithLogger` with `knative/pkg/logging.WithLogger`, or are they here for any particular reason?